### PR TITLE
[POAE7-2811] Support continue in LoopBuilder.

### PR DESCRIPTION
**Pull request by bigPYJ1151**
### What changes were proposed in this pull request?

Support continue semantic for LoopBuilder.

### Why are the changes needed?

Support Left Join

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs

### Which label does this PR belong to?

INFRA

### CODE REVIEW

**Comment by bigPYJ1151** [comment]
Pls help to review, thx~ @winningsix @Ma-Jian1 @qiuyangshen

File: src/cider/tests/nextgen/jitlib/JITLibTest.cpp [hunk 1]
```diff
@@ -540,6 +540,41 @@ TEST_F(JITLibTests, NestedForControlFlowTest) {
...
+          .addReturn(JITTypeTag::INT32)
+          .addProcedureBuilder([](JITFunctionPointer function) {
+            JITValuePointer ret = function->createVariable(JITTypeTag::INT32, "ret", 0);
+
+            auto loop_builder = function->createLoopBuilder();
+            JITValuePointer index =
+                function->createVariable(JITTypeTag::INT32, "index", 10);
+
+            loop_builder->condition([&]() { return index + 0; })
+                ->loop([&](LoopBuilder*) {
```
Comment (Ma-Jian1) [inline src/cider/tests/nextgen/jitlib/JITLibTest.cpp:558]: seems you capture the loop_builder. then why do you need LoopBuilder* arguments?

File: src/cider/tests/nextgen/jitlib/JITLibTest.cpp [hunk 2]
```diff
@@ -540,6 +540,41 @@ TEST_F(JITLibTests, NestedForControlFlowTest) {
...
+                  function->createIfBuilder()
+                      ->condition([&index]() { return index % 2 != 0; })
+                      ->ifTrue([&]() { loop_builder->loopContinue(); })
```
Comment (Ma-Jian1) [inline src/cider/tests/nextgen/jitlib/JITLibTest.cpp:561]: condition as argument of the loopContinue() is easier to use?
```cpp
loop_builder->loopContinue( index % 2 != 0 );
```

**Comment by qiuyangshen** [comment]
LGTM

**Review by Ma-Jian1 — APPROVED** [review]

**Review by Ma-Jian1 — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**